### PR TITLE
Add 'azure-storage-explorer'

### PR DESCRIPTION
**Pull request by jazzwang**
- [X] Checked there aren’t open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [X] Checked there aren’t closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [X] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [X] Commit message includes cask’s name.
- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` left no offenses.
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.

### CODE REVIEW

File: Casks/azure-storage-explorer.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask 'azure-storage-explorer' do
+  version '0.7.20160509.0'
+  sha256 :no_check
+
+  url 'http://go.microsoft.com/fwlink/?LinkId=708342'
+  name 'Microsoft Azure Storage Explorer'
+  homepage 'http://storageexplorer.com/'
+  license # unkown
```
Comment (vitorgalvao) [inline Casks/azure-storage-explorer.rb]: This cannot have passed audit. This cannot be empty.

File: Casks/azure-storage-explorer.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cask 'azure-storage-explorer' do
+  version '0.7.20160509.0'
```
Comment (vitorgalvao) [inline Casks/azure-storage-explorer.rb]: Seeing as `url` doesn’t point to a specific version, this cask should use `version :latest` and `sha256 :no_check`. [See an example](https://github.com/caskroom/homebrew-cask/blob/f25b6babcd398abf48e33af3d887b2d00de1d661/Casks/kibako.rb).

File: Casks/azure-storage-explorer.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+cask 'azure-storage-explorer' do
```
Comment (vitorgalvao) [inline Casks/azure-storage-explorer.rb]: Token reference was not followed. This must be `microsoft-azure-storage-explorer`.

**Comment by jazzwang** [comment]
Thanks for the review of @vitorgalvao . 
- I'll try to follow `Token reference`, use `microsoft-azure-storage-explorer` instead.
- use `version :latest` instead.
- use `license :gratis` instead.

**Comment by vitorgalvao** [comment]
Please @mention me after you do, so I get an email to check back.

**Comment by vitorgalvao** [comment]
Closing and opening does not affect Travis. You still haven’t made the fixes.

**Pull request closed without merging**